src/util/address.rs

```diff
@@ -204,6 +204,15 @@ impl Address {
         Address(self.0 + size)
     }
 
+    // We implemented the Sub trait but we still keep this sub function.
```

Comment:
Have you tested that the non-const `Sub` trait doesn't work for Address constants?
(E.g. we can use the `Sub` trait for int constants)

Reply:
Umm, it seems we can. Thanks. We need to include `#![feature(const_trait_impl)]` and `impl const Sub for Address {}`. I will change this.

Reply:
I gave this a try, but `const_trait_impl` seems buggy at the moment. I can replace our definition for `Address` constants  in mmtk-core with `+` or `-`, but I can't do that for our binding crates (I have to use `const fn add/sub()` in the bindings). It could be a bug in nightly rust, and I have submitted an issue for them (https://github.com/rust-lang/rust/issues/84846). Probably for now, we have to keep both const traits and the const function `add/sub()`.

Reply:
It seems if I declare a constant (e.g. `AVAILABLE_START = HEAP_START + VM_SPACE_SIZE`), and a binding uses `AVAILABLE_START`, the compiler also gives an error. This means we can't use this trait for now. @javadamiri

Reply:
Right